conans/server/crypto/jwt/jwt_manager.py

```diff
@@ -22,7 +24,12 @@ def get_token_for(self, profile_fields=None):
         if self.expire_time:
             profile_fields["exp"] = datetime.utcnow() + self.expire_time
 
-        return jwt.encode(profile_fields, self.secret, algorithm="HS256")
+        encoded = jwt.encode(profile_fields, self.secret, algorithm="HS256")
+        if self.expire_time and isinstance(profile_fields.get("exp"), datetime):
```

Comment:
Passing this after the ``jwt.encode()`` reads confusing or even dangerous? What if ``jwt.encode`` does an internal copy? This field would be ignored, right? Is this extracted from jwt documentation?

Reply:
Is this in addition to the above ``profile_fields["exp"] = datetime.utcnow() + self.expire_time``? 

I might be worth commenting a bit these changes, what the code does is not obvious.

Reply:
>  Is this extracted from jwt documentation?

No, I copied directly from their source code, to keep the compatibility: https://github.com/jpadilla/pyjwt/compare/1.4.0...2.1.0#diff-6893ad4a1c5a36b8af3028db8c8bc3b62418149843fc382faf901eaab008e380R53

>  What if jwt.encode does an internal copy?

Indeed there is a copy for payload. Let me invert it.

Reply:
Done, I've added some comments. Indeed won't remember about these changes few weeks later, so comments a really helpful here.

I also inverted the encoder order, as profile_fields is copied internally for 2.x, we don't want to risk any possible abnormal behavior for future minor versions.